### PR TITLE
fix: use connection name for the Databases tab COMPASS-8049

### PR DESCRIPTION
**Pull request by paula-stacho**
## Description
Moved the icon picking from the sidebar to compass-connections, so that the logic is in a single place. The status marker is not displayed in the tabs at the moment, but it's something we'll likely add later (explicit disconnect closes the tabs, and as of now we don't check if a connection is stale, so it would be just unreliable green markers everywhere).

Screenshot 2024-07-17 at 12 50 07

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-workspaces/src/components/workspaces.tsx [hunk 1]
```diff
@@ -181,15 +185,17 @@ const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
...
-        case 'Databases':
+        case 'Databases': {
+          const connectionName = getConnectionTitleById(tab.connectionId);
           return {
             id: tab.id,
-            connectionName: getConnectionTitleById(tab.connectionId),
+            connectionName,
             type: tab.type,
-            title: tab.type,
+            title: multiConnectionsEnabled ? connectionName : tab.type,
```
Comment (gribnoysup) [inline packages/compass-workspaces/src/components/workspaces.tsx:194]: We decided already that it's okay to have the same breadcrumb for this tab without the feature flag check, why are we introducing a difference here? Is it a requirement?

File: packages/compass-workspaces/src/components/workspaces.tsx [hunk 2]
```diff
@@ -181,15 +185,17 @@ const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
...
             iconGlyph: 'Database',
```
Comment (gribnoysup) [inline packages/compass-workspaces/src/components/workspaces.tsx:195]: Ticket also mentions changing the icon here in addition to the title

Comment (paula-stacho) [reply packages/compass-workspaces/src/components/workspaces.tsx:195]: yes, this is work in progress, I was waiting to get some question around the icon answered

File: packages/compass-workspaces/src/components/workspaces.tsx (hunk 1 above)
Comment (paula-stacho) [reply packages/compass-workspaces/src/components/workspaces.tsx:194]: when you say breadcrumb I think of subtitle, not the title. I don't recall us deciding on changing the title for all, but maybe I missed that for the same reason 😅 let me check on the channel

**Comment by paula-stacho** [comment]
update: putting this to block as we might change the design

**Pull request closed without merging**